test/suites/private/getEnvironment.m

```diff
@@ -1,25 +1,25 @@
 function [env, versionString] = getEnvironment()
-% Checks if we are in MATLAB or Octave.
+% Determine environment (Octave, MATLAB) and version string
+% TODO: Unify the two pivate functions!
     persistent cache
-    
-    alternatives = {'MATLAB', 'Octave'};
+
     if isempty(cache)
-        for iCase = 1:numel(alternatives)
-            env   = alternatives{iCase};
+        isOctave = exist('OCTAVE_VERSION', 'builtin') ~= 0;
+        if isOctave
+            env = 'Octave';
+            versionString = OCTAVE_VERSION;
+        else
+            env = 'MATLAB';
             vData = ver(env);
-            if ~isempty(vData) % found the right environment
-                versionString = vData.Version;
-                % store in cache
-                cache.env = env;
-                cache.versionString = versionString;
-                return;
-            end
+            versionString = vData.Version;
         end
-        % fall-back values
-        env = '';
-        versionString = '';
+
+        % store in cache
+        cache.env = env;
+        cache.versionString = versionString;
+
```

Comment:
I think it's certainly possible to assign immediately to `cache`, no noticeable implications (at runtime) of doing it one way or another.

I do think, however, that this implementation is slightly clearer: 
1. determine the env + version,
2. store that in cache

instead of mixing cache and logic together.